src/main/java/com/example/demo/productComponent/domain/services/ProductService.java

```diff
@@ -0,0 +1,199 @@
+import com.example.demo.productComponent.api.dtos.PostProductDTO;
+import com.example.demo.productComponent.api.dtos.ProductAndModifierHelperDTOs.ProductDTO;
+import com.example.demo.productComponent.api.dtos.ResponseProductDTO;
+import com.example.demo.productComponent.domain.entities.Product;
+import com.example.demo.productComponent.repository.ProductCompatibleModifierRepository;
+import com.example.demo.productComponent.repository.ProductRepository;
+import jakarta.persistence.OptimisticLockException;
+import lombok.AllArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.UUID;
+
+@AllArgsConstructor
+@Service
+public class ProductService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProductService.class);
+
+    private final ProductRepository productRepository;
+    private final ProductValidator productValidator;
+    private final ProductCompatibleModifierRepository productCompatibleModifierRepository;
+
+    @Transactional
+    public ProductDTO createProduct(PostProductDTO postProductDTO){
+        // Validate the post product DTO
+        productValidator.validatePostProductDTO(postProductDTO);
+
+        // Make sure that all the product modifiers exist
+        validateModifiers(postProductDTO.getCompatibleModifierIds());
+
+        // Map the DTO to the model and save it
+        Product product = Mapper.mapToModel(postProductDTO, ProductMapper.TO_MODEL);
+        Product savedProduct = productRepository.save(product);
+
+        // Add the product modifiers to the product
+        postProductDTO.getCompatibleModifierIds().forEach(modifierId ->
+            addModifierToProduct(savedProduct.getId(), modifierId)
+        );
+
+        // Save the product and return the DTO
+        return Mapper.mapToDTO(savedProduct, ProductMapper.TO_DTO);
+    }
+
+    public GetProductsDTO getProductsByBusinessId(UUID businessId){
+        // Get all the products by business id, map them to DTOs
+        List<ProductDTO> productDTOS = productRepository.findAllByBusinessId(businessId).stream()
+                .map(ProductMapper.TO_DTO::map)
+                .toList();
+
+        return GetProductsDTO.builder()
+                .items(productDTOS)
+                .businessId(businessId)
+                .totalItems(productDTOS.size())
+                .build();
+    }
+
+    public GetProductsDTO getProductsByBusinessId(int page, int size, UUID businessId){
+        // Get all the products by business id, map them to DTOs
+        Page<ProductDTO> productDTOS = productRepository.findAllByBusinessId(businessId, PageRequest.of(page, size))
+                .map(ProductMapper.TO_DTO::map);
+
+        return GetProductsDTO.builder()
```

Comment:
Nezinau man cia keistai irgi atrodo, lygtais mapperis parasomas, man jau blogai po 1000 ar kiek cia perziuretu eiluciu

Reply:
cia blogai, pakeiciau

Reply:
Write function to mapper